pallets/dapp-staking-v3/src/lib.rs

```diff
@@ -1584,13 +1629,14 @@ pub mod pallet {
 
         /// Returns the number of blocks per voting period.
         pub(crate) fn blocks_per_voting_period() -> BlockNumberFor<T> {
-            T::StandardEraLength::get().saturating_mul(T::StandardErasPerVotingPeriod::get().into())
+            T::StandardEraLength::get()
+                .saturating_mul(T::StandardErasPerVotingSubperiod::get().into())
         }
 
         /// `true` if smart contract is active, `false` if it has been unregistered.
         pub(crate) fn is_active(smart_contract: &T::SmartContract) -> bool {
             IntegratedDApps::<T>::get(smart_contract)
-                .map_or(false, |dapp_info| dapp_info.state == DAppState::Registered)
+                .map_or(false, |dapp_info| dapp_info.is_active())
```

Comment:
Probably could be improved later, the `active` and `IntegratedDApps` containing states might be a bit confusing. We can either have a clearer naming of each different state or have terminology explained in header docstring.

Reply:
Sorry, I'm not sure I get the comment - would you like me to remain `is_active`? Or remove the `state` from `DAppInfo`?

Reply:
Let me try to elaborate. There are currently three cases for smart contract state check
- Check if `is_active`
- Check if `IntegratedDApps` contains the contract
- Check both

It's probably just me, the terms `active` (it's actually a registration/listing check) and `integrated` are a bit confusing to understand until digging deeper into the implementation. IMO this could be improved by either more explicit naming or adding docs to cover the items.

(A big but) BUT let's merge the PR now without any change and address it later in the main dApp staking PR.

Reply:
Oh, I got it! Yeah, let's address this later. I will add a TODO so I don't forget about it.